core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -56,6 +58,11 @@ public CustomFieldQuery(Query query, IndexReader reader, boolean phraseHighlight
 
     @Override
     void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQueries, float boost) throws IOException {
+        while (sourceQuery instanceof BoostQuery) {
```

Comment:
Yes, After investigation, there is no directly `BoostQueryBuilder` api, so it should not have `multiple` layers `BootQuery`. 

If in future there are multi layer `BoostQuery`, I think this should be call out. :)